### ISSUE TITLE
Error when starting diff mode: E788: Not allowed to edit another buffer now

### ISSUE THREAD

**Issue by Coacher** [rickhowe/diffchar.vim#7]
Hello.

vim-8.0.739, diffchar-6.7.

If I start vimdiff with 2 files or open two files in buffers and do `diffthis` in both I get:
```
Error detected while processing function diffchar#SetDiffModeSync[6]..<SNR>122_ValidDiffModeWins:
line    5:
E788: Not allowed to edit another buffer now
E788: Not allowed to edit another buffer now
line   14:
E788: Not allowed to edit another buffer now
```

Same with diffchar-6.6, 6.5. AFAIU a recent change in vim broke diffchar plugin.

**Comment by Coacher** [comment]
vim --version: https://paste.pound-python.org/show/6DjKmYRtFE65V240vcqk/

**Comment by rickhowe** [comment]
Thank you. I also found this regression on patch 736. I need to look into the cause but please try to add the following 3 lines in plugin/diffchar.vim.
```
@@ -1196,6 +1196,9 @@
 endfunction
 
 function! s:ValidDiffModeWins(wlist)
+	if has('patch-8.0.736')
+		return filter(copy(a:wlist), 'getwinvar(v:val, "&diff")')
+	endif
 	" try to use diffput to check if the diff mode is really valid
 	let cwin = winnr()
 	let vdmw = []
```

**Comment by Coacher** [comment]
Your patch removes the error.

**Comment by Coacher** [comment]
diffchar-6.8 conflicts with this patch. The issue is no longer reproducible with diffchar-6.8. Thank you!

I can see diffchar-6.8 in my local clone of diffchar managed by neobundle, but not in GH commit log.
Force pushes? Not closing the issue yet because of this inconsistency.

**Comment by rickhowe** [comment]
Yes, your 6.8 should fix E788 but I rejected it from the Git repository to add another fix. Thank you.

**Comment by Coacher** [comment]
> Yes, your 6.8 should fix E788 but I rejected it from the Git repository to add another fix.

Now updating requires manual intervention. Not cool.

**Comment by Coacher** [comment]
Appears to be fixed in 6.8. Closing. Thank you.
